frontend/hooks/usePlutoDesktop.js

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "../imports/Preact.js"
+
+/**
+ * @typedef {(window: Window) => void} DesktopHookCallback
+ */
+
+/**
+ * A hook to check whether pluto is running inside Desktop,
+ * and execute a callback if that is the case.
+ * @param {DesktopHookCallback | undefined} callback A function that takes in a window (having `plutoDesktop` as a property)
+ * as its argument, if `plutoDesktop` is not found, it is not called. **DO NOT PASS ANY FUNCTION THAT CHANGES STATE OF THE COMPONENT.**
+ * @returns A boolean denoting whether pluto is running inside Desktop or not and a condtional callback utility function.
+ */
+const usePlutoDesktop = (/** @type {DesktopHookCallback | undefined} */ callback) => {
+    const [isDesktop, setIsDesktop] = useState(false)
```

Comment:
You can use `const isDesktop = window.plutoDesktop != null`, right?

And then

```js
useEffect(() => {
        if (isDesktop) {
            // console.log(
            //   'Running in Desktop Environment! Found following properties/methods:',
            //   window.plutoDesktop
            // );
            if (callback) callback(window)
        }
    }, [callback])
```

Reply:
Actually, maybe the `usePlutoDesktop` is not really necessary? You could replace this entire file with:

```js
export const isDesktop = window.plutoDesktop != null
```

And just use a `useEffect` in ExportBanner.js

Reply:
Ah yes, this work too! Committed this!

Reply:
> Actually, maybe the `usePlutoDesktop` is not really necessary? You could replace this entire file with:
> 
> ```js
> export const isDesktop = window.plutoDesktop != null
> ```
> 
> And just use a `useEffect` in ExportBanner.js

Yeah I added for any possible future usage, although yes, we can remove it altogether and it would be just fine.

Reply:
Nice, let's remove it for now. 

100% of bugs are caused by code, so less code = less bugs

![](https://media.giphy.com/media/W3a0zO282fuBpsqqyD/giphy.gif)

Reply:
>
> 100% of bugs are caused by code, so less code = less bugs
>
Haven't read anything truer.
Fixed it :)